### PR TITLE
Chat inbox perf fixes

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers This is based on my user timings stuff.

1. Sort was being called a lot and its very slow due to immutable-js records. its actually faster to convert the whole thing into simple objects then sorting :(
1. The request to delay handing unboxing due to idle timing with a 100ms timeout actually resolves the timeout waaaay later (like 1 s) due to all the work we're doing in parallel. Instead of having all these unbox calls fighting each other we just have a single simple queue

### CODE REVIEW

File: shared/actions/chat/inbox.js [hunk 1]
```diff
@@ -274,20 +274,31 @@ function* untrustedInboxVisible(action: Constants.UntrustedInboxVisible): SagaGe
...
+  yield put(Creators.unboxMore())
   return EngineRpc.rpcResult()
 }
 
+function* unboxMore(): SagaGenerator<any, any> {
+  if (!_chatInboxToProcess.length) {
+    return
+  }
+
+  const conv = _chatInboxToProcess.pop()
```
Comment (akalin-keybase) [inline shared/actions/chat/inbox.js]: did you mean `shift`? Currently this behaves like a stack, but it sounds like you intended a queue.

File: shared/chat/inbox/container.js [hunk 2]
```diff
@@ -47,6 +47,10 @@ const getSimpleRows = createSelector(
 
         return !isEmpty && !isSuperseded && !isFilteredOut
       })
+      .map(i => ({
```
Comment (akalin-keybase) [inline shared/chat/inbox/container.js]: This probably warrants a comment explaining that this is done for perf reasons and that sorting immutable lists is slow, etc.

File: shared/actions/chat/inbox.js (hunk 1 above)
Comment (chrisnojima) [reply shared/actions/chat/inbox.js]: no i meant pop as i figured the most recent thing you asked for is likely what you want (aka scrolling and etc)

File: shared/chat/inbox/container.js (hunk 2 above)
Comment (chrisnojima) [reply shared/chat/inbox/container.js]: sounds good

**Pull request merged**